services/worker/src/worker/job_runner.py

```diff
@@ -1,322 +1,20 @@
-            # Add lines in the traceback to give some info about the previous step error (a bit hacky)
-            "cause_traceback": [
-                "The previous step failed, the error is copied to this step:",
-                f"  {kind=} {dataset=} {config=} {split=}",
-                "---",
-            ],
-        }
-        if "cause_exception" in response["details"] and isinstance(response["details"]["cause_exception"], str):
-            error_with_cause["cause_exception"] = response["details"]["cause_exception"]
-        if "cause_message" in response["details"] and isinstance(response["details"]["cause_message"], str):
-            error_with_cause["cause_message"] = response["details"]["cause_message"]
-        if (
-            "cause_traceback" in response["details"]
-            and isinstance(response["details"]["cause_traceback"], list)
-            and all(isinstance(line, str) for line in response["details"]["cause_traceback"])
-        ):
-            error_with_cause["cause_traceback"].extend(response["details"]["cause_traceback"])
-        return PreviousStepError(
-            message=message,
-            status_code=status_code,
-            code=error_code,
-            cause=cause,
-            disclose_cause=disclose_cause,
-            error_without_cause=error_without_cause,
-            error_with_cause=error_with_cause,
-        )
-
-    def as_response_with_cause(self) -> ErrorResponseWithCause:
-        return self.error_with_cause
-
-    def as_response_without_cause(self) -> ErrorResponseWithoutCause:
-        return self.error_without_cause
-
-
-def get_previous_step_or_raise(
-    kinds: List[str], dataset: str, config: Optional[str] = None, split: Optional[str] = None
-) -> BestResponse:
-    """Get the previous step from the cache, or raise an exception if it failed."""
-    best_response = get_best_response(kinds=kinds, dataset=dataset, config=config, split=split)
-    if best_response.response["http_status"] != HTTPStatus.OK:
-        raise PreviousStepError.from_response(
-            response=best_response.response,
-            kind=best_response.kind,
-            dataset=dataset,
-            config=config,
-            split=split,
-        )
-    return best_response
+from worker.config import AppConfig
+from worker.utils import JobResult, JobRunnerInfo
 
 
 class JobRunner(ABC):
-    """
-    Base class for job runners. A job runner is a class that processes a job, for a specific processing step.
-
-    It cannot be instantiated directly, but must be subclassed.
-
-    Args:
-        job_info (:obj:`JobInfo`):
-            The job to process. It contains the job_id, the job type, the dataset, the config, the split
-            the force flag, and the priority level.
-        common_config (:obj:`CommonConfig`):
-            The common config.
-        processing_step (:obj:`ProcessingStep`):
-            The processing step to process.
-    """
-
-    job_id: str
```

Comment:
why to you remove the fields that are initialized in __init__?

Reply:
Most of them were moved to job_manager

Reply:
but you're still initializing them in `__init__`: https://github.com/huggingface/datasets-server/pull/1146/files/f96451eed128bc531ea383b98d4e62e550413d0c#diff-7ccf0bcf2a0d7e66a9eae13d7168a08a4b883b8792b418a0a60cd86d7f0e8d7fR37

Reply:
they should not be initialized in job_runner I think.

Reply:
Totally agree, sorry I misunderstood.
I will fix it on https://github.com/huggingface/datasets-server/pull/1170